### PR TITLE
Document link interception details

### PR DESCRIPTION
**Pull request by alshakero**
Related https://github.com/Palindrom/Palindrom/issues/73 & https://github.com/Palindrom/Palindrom/issues/70

### CODE REVIEW

File: docs/05-Navigation Interception.md [hunk 1]
```diff
@@ -0,0 +1,16 @@
+## Navigation Interception
+
+By default, PalindromDOM intercepts browser navigation and sends an AJAX request to the server for the next page's partial/model. This results in a more efficient and faster navigation.
```
Comment (warpech) [inline docs/05-Navigation Interception.md]: The word "partial" does not exist in Palindrom docs by now. For a very good reason: Palindrom is not only meant for Starcounter views, but in general, for "two-way data binding between local and remote JSON models".

I propose to rewrite this sentence as following:

```
PalindromDOM gives a regular HTML document the benefits of a single-page application 
by adding a special optimisation to the internal web links. 

Regular links in the current host are cancelled and replaced by a GET  `XmlHttpRequest` to 
the same URL with `Accept: application/json-patch+json`. Upon this request, the server 
modifies the model and responds with a patch that contain the minimum set of changes 
to navigate to the URL.

The browser navigation (i.e. the back, forward and reload buttons) is preserved by the use 
of the HTML History API.

Only links in the current host without the target attribute or with the target attribute set 
to `_self` are intercepted.
```

File: docs/05-Navigation Interception.md [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+
+PalindromDOM only intercepts links that are part of your application and doesn't intercept external links or links that have a `target` set to anything except `self`.
+
+### How to disable interception
+
+Sometimes, you need to disable interception even for links within the application. And as mentioned above, If you set any `target` that doesn't equal `self` to your link, the navigation from that anchor will not be intercepted.
```
Comment (warpech) [inline docs/05-Navigation Interception.md]: > above, If you set any `target` that doesn't equal `self` to your link, the

above, if you set the `target` attribute to anything else than `_self`, the

File: docs/05-Navigation Interception.md [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+Example:
+
+```html
+<a href="internalLink.html" target="hello">Go to internal link></a>
```
Comment (warpech) [inline docs/05-Navigation Interception.md]: target="_top"

File: docs/05-Navigation Interception.md [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+```
+... even if `internalLink.html` is part of your application, PalindromDOM will not intercept the navigation and the browser will issue a full HTTP request to `internalLink.html`.
```
Comment (warpech) [inline docs/05-Navigation Interception.md]: As the value of the `target` attribute, you might consider one of the following keywords that have a special meaning in HTML:

- `_top` - load the URL into the current browser tab or window
- `_blank` - load the URL into a new browser tab or window

You can find more information on the `target` attribute at [MDN](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a).

**Comment by alshakero** [comment]
> that's a bug in PalindromDOM that I overlooked when reviewing #145. Sorry for that :(

No issue at all! Should be fixed in https://github.com/Palindrom/Palindrom/commit/1f62d6738641930961c4b49631b22e4998834b77

**Pull request merged**